Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -929,60 +940,65 @@ class NexusGeometrySaveImpl {
   }
 }; // class NexusGeometrySaveImpl
 
-/*
- * Function: saveInstrument
+/* Internal-use only signature:
+ * Function: _saveInstrument
  * calls the save methods to write components to file after exception
  * checking. Produces a Nexus format file containing the Instrument geometry
  * and metadata.
  *
  * @param compInfo : componentInfo object.
  * @param detInfo : detectorInfo object.
- * @param fullPath : save destination as full path.
- * @param rootName : name of root entry
+ * @param detId2IndexMap: (optional) detid2index_map
+ * @param parentGroup : open H5::Group in which to place the NXinstrument.
  * @param logger : logging object
- * @param append : append mode, means openting and appending to existing file.
- * If false, creates new file.
  * @param reporter : (optional) report to progressBase.
  */
-void saveInstrument(const Geometry::ComponentInfo &compInfo, const Geometry::DetectorInfo &detInfo,
-                    const std::string &fullPath, const std::string &rootName, AbstractLogger &logger, bool append,
-                    Kernel::ProgressBase *reporter) {
 
-  validateInputs(logger, fullPath, compInfo);
-  // IDs of all detectors in Instrument
-  H5::Group rootGroup;
-  H5::H5File file;
-  if (append) {
-    file = H5::H5File(fullPath, H5F_ACC_RDWR); // open file
-    rootGroup = file.openGroup(rootName);
-  } else {
-    file = H5::H5File(fullPath, H5F_ACC_TRUNC); // open file
-    rootGroup = file.createGroup(rootName);
```

Comment:
And this is handled somewhere else now, right?  Since we dont push everything to root but instead the bank.